### DISCUSSION TITLE
React Native - Error opening socket

### DISCUSSION THREAD

**Discussion by egueiros** [WalletConnect/walletconnect-monorepo#894] [Developer Support]
Hello, I'm following the quick start guide for 2.0 for React Native (https://docs.walletconnect.com/2.0/quick-start/wallets/react-native). Currently it's failing to connect with the following error. 

```
TypeError: Cannot read properties of undefined (reading 'message')
    at parseConnectionError (error.js:50:14)
    at WsConnection.parseError (ws.js:117:16)
    at WebSocket.socket.onerror (ws.js:81:36)
    at WebSocket.dispatchEvent (event-target-shim.js:818:35)
    at WebSocket.js:271:14
    at EventEmitter.emit (_EventEmitter.js:149:33)
    at MessageQueue.__callFunction (MessageQueue.js:419:27)
    at MessageQueue.js:116:12
    at MessageQueue.__guard (MessageQueue.js:370:9)
    at MessageQueue.callFunctionReturnFlushedQueue (MessageQueue.js:115:10)
```
I traced it to `relayer.js -> init() -> provider.connect()` then ` provider.js -> connect() -> open(connection)`. After this I get the error and the client does not init. 
    
Currently my dependencies are:

```
"@walletconnect/client": "^2.0.0-beta.26",
"@walletconnect/types": "^1.7.7",
"@react-native-async-storage/async-storage": "^1.17.2",
```

Don't know if this has anything to do with the issue, but `storageOptions` complains about `AsyncStorage`. 
![Screen Shot 2022-03-30 at 9 48 56 PM](https://user-images.githubusercontent.com/1929616/160972683-f7eff637-3310-46e7-a1b6-4d193ef213f0.png)

Is there a specific version of AsyncStorage that will conform with this interface and make the error go away, or that should not matter?

Below is my complete init options
```
 walletConnectClient = await WalletConnectClient.init({
    controller: true,
    projectId: 'MY_PROJECT_ID',
    relayUrl: 'wss://relay.walletconnect.com',
    metadata: {
      name: 'React Wallet',
      description: 'React Wallet for WalletConnect',
      url: 'https://walletconnect.com/',
      icons: ['https://avatars.githubusercontent.com/u/37784886']
    },
    storageOptions: {
      asyncStorage: AsyncStorage,
    }
  })
```
Any insight in this is greatly appreciated.

**Comment by xzilja** [comment]
I think storage related issue is purely types, will need to look into that. As for the actual error it looks like something was thrown during connection with error message being undefined.

Quick Q, did you replace `'MY_PROJECT_ID'` with your actual project id?

If you did and it's still not working, I'll have to spin up rn project and test myself 👍 
Sorry for this, we are currently working on overhauling the docs (will include better rn guide) as well as making examples. One of them will also be rn example at some point.

**Reply by egueiros** [reply]
Yes, project id would have the id of the project created in WalletConnect. I just have a simple project right now that I can put on GitHub. Let me do that and I'll share the link here. 

Thanks for looking into this.

**Comment by egueiros** [comment]
Here's the link to my simple project. https://github.com/egueiros/rn-walletconnect

Just messing around more with my environment...if I hotspot my dev machine and use cellular data, it connects. Something with my WiFi? Not sure, other sockets connect ok. We'll keep investigating. 

When I was able to init the client, unfortunately it was threw a new error now when trying to pair. 
<img width="385" alt="Screen Shot 2022-03-31 at 9 21 00 AM" src="https://user-images.githubusercontent.com/1929616/161094307-37d1c033-0fe2-42f4-99c4-6941aec71286.png">

**Reply by egueiros** [reply]
seems like under `parseUri` in **@walletconnect/utils**, right here
```
 const result = {
        protocol,
        topic: requiredValues[0],
        version: parseInt(requiredValues[1], 10),
        publicKey: queryParams.publicKey,
        controller: queryParams.controller === "true",
        relay: safeJsonParse(queryParams.relay),
    };
```
it's expecting a `publicKey`, `controller`, and `relay` query params. But the ParaSwap example gives `bridge` and `key` params
